src/silhouette.jl

```diff
@@ -81,24 +80,24 @@ function silhouettes(assignments::AbstractVector{<:Integer},
         a[j] = r[l, j]
 
         v = S(Inf)
-        p = -1
         for i = 1:k
             @inbounds rij = r[i,j]
-            if (i != l) && (rij < v)
-                v = rij
-                p = i
-            end
+            i != l && rij < v && (v = rij)
         end
-        b[j] = v
+        b[j] = v 
     end
 
     # compute silhouette score
     sil = a   # reuse the memory of a for sil
     for j = 1:n
         if counts[assignments[j]] == 1
-            sil[j] = 0
+            sil[j] = zero(S)
         else
-            @inbounds sil[j] = (b[j] - a[j]) / max(a[j], b[j])
+            # b[j] and a[j] can be Inf so best to ensure Inf/Inf division
+            # is avoided.
+            @inbounds sil[j] = a[j] < b[j] ? one(S) - a[j]/b[j] :
+                               a[j] > b[j] ? b[j]/a[j] - one(S) :
+                               zero(S) 
```

Comment:
I've just tested: Julia figures out the correct type for `1` at compile time. Please also fix the trailing space at the last line

```julia
            @inbounds sil[j] = a[j] < b[j] ? 1 - a[j]/b[j] :
                               a[j] > b[j] ? b[j]/a[j] - 1 :
                               zero(eltype(r))
```